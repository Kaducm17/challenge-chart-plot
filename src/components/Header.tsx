import React, { CSSProperties } from 'react'

interface Props {
  name: string
}

const headerStyle = {
  background: '#cccc',
  padding: '10px'
} as CSSProperties

const Header = ({ name }: Props) => {
  return (
    <section style={headerStyle}>
      <h2>{name}'s Challenge</h2> 
    </section>
  )
}

export default Header