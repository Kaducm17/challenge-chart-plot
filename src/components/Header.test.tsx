import {create, act} from 'react-test-renderer';
import Header from '../components/Header'; // The component being tested

// render the component
it("Will test if header receives the name props", () => {
	let root; 
	act(() => {
	  root = create(<Header name='Kadu'/>)
	});

	// make assertions on root 
	expect(root.toJSON()).toMatchSnapshot();

	// update with some different props
	act(() => {
	  root.update(<Header name='Intelie'/>);
	})

	// make assertions on root 
	expect(root.toJSON()).toMatchSnapshot();
});