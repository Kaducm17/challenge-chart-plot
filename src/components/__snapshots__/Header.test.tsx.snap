// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<section
  style={
    Object {
      "background": "#cccc",
      "padding": "10px",
    }
  }
>
  <h2>
    Kadu
    's Challenge
  </h2>
</section>
`;

exports[` 2`] = `
<section
  style={
    Object {
      "background": "#cccc",
      "padding": "10px",
    }
  }
>
  <h2>
    Intelie
    's Challenge
  </h2>
</section>
`;

exports[`Will test if header receives the name props 1`] = `
<section
  style={
    Object {
      "background": "#cccc",
      "padding": "10px",
    }
  }
>
  <h2>
    Kadu
    's Challenge
  </h2>
</section>
`;

exports[`Will test if header receives the name props 2`] = `
<section
  style={
    Object {
      "background": "#cccc",
      "padding": "10px",
    }
  }
>
  <h2>
    Intelie
    's Challenge
  </h2>
</section>
`;
